lib/Avatar/AvatarWithPopover.js

```diff
@@ -16,57 +16,106 @@ import { Avatar, Icon } from '../';
  * </AvatarWithPopover>
  */
 class AvatarWithPopover extends Component {
+  constructor(props) {
+    super(props);
+    this.togglePopover = this.togglePopover.bind(this);
+    this.hidePopover = this.hidePopover.bind(this);
+    this.showPopover = this.showPopover.bind(this);
+  }
   state = {
     showing: false
   };
 
-  render() {
-    const popover = (
-      <Popover id={this.props.email} className={this.props.popoverClass}>
-        {this.props.children}
-      </Popover>
-    );
+  togglePopover() {
+    this.setState({ showing: !this.state.showing });
+  }
 
-    const onEnter = () => {
-      this.setState({ showing: true });
-    };
-    const onExited = () => {
-      this.setState({ showing: false });
-    };
+  hidePopover() {
+    this.setState({ showing: false });
+  }
 
+  showPopover() {
+    this.setState({ showing: true });
+  }
+
+  renderTrigger() {
     const classes = cx('button button--icon-only avatar__wrapper', {
       'is-active': this.state.showing
     });
 
+    const props = {};
+    if (typeof this.props.triggerEvent === 'string') {
+      props[this.props.triggerEvent] = this.togglePopover;
+    } else {
+      this.props.triggerEvent.forEach(event => {
+        if (event === 'onHover') {
+          props.onMouseEnter = this.showPopover;
+          props.onMouseLeave = this.hidePopover;
+        } else {
```

Comment:
Try to refrain from using `else` for various reasons. Laracast did a decent video for this a while back: https://laracasts.com/series/simple-rules-for-simpler-code/episodes/2

By moving the logic into a method (previous feedback) I believe it should be quite simple and it'll make the logic really easy to understand e.g.

```
if (event === 'onHover') {
  return {
    onMouseEnter: this.showPopover,
    onMouseLeave: this.hidePopover
  }
}

return {
  [event]: this.togglePopover
}
```

Reply:
Done these changes mate